### PR TITLE
JAVA-1714: Make replication strategies pluggable

### PR DESCRIPTION
**Pull request by olim7t**

### CODE REVIEW

File: core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/DefaultReplicationStrategyFactory.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+public class DefaultReplicationStrategyFactory implements ReplicationStrategyFactory {
+
+  private final String logPrefix;
+
+  public DefaultReplicationStrategyFactory(InternalDriverContext context) {
+    this.logPrefix = context.clusterName();
+  }
+
+  @Override
+  public ReplicationStrategy newInstance(Map<String, String> replicationConfig) {
```
Comment (tolbertam) [inline core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/DefaultReplicationStrategyFactory.java:31]: It'd be nice to have some implementation of this that doesn't throw an IllegalArgumentException when no match is found, that way one could reuse this logic to evaluate these strategies and than add their additional code for additional strategies they implement.

**Review by tolbertam — APPROVED** [review]
Had one small comment, but otherwise this lgtm :+1:

File: core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/DefaultReplicationStrategyFactory.java (hunk 1 above)
Comment (olim7t) [reply core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/DefaultReplicationStrategyFactory.java:31]: If you test your strategies first and only delegate to `super()` if none matches, it should work no?

Comment (tolbertam) [reply core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/DefaultReplicationStrategyFactory.java:31]: Oh hah, can't believe I didn't think of that, yeah that does work great :+1:

**Pull request merged**